### PR TITLE
PIN-1591: fix QueueHandler.processMessage Future usage

### PR DESCRIPTION
**Pull request by beetlecrunch**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/notifier/service/impl/QueueHandler.scala [hunk 1]
```diff
@@ -47,8 +47,9 @@ class QueueHandler(
     _ = logger.debug("Organization id retrieved for message {} -> {}", msg.messageUUID, organizationId)
     nextEvent <- idRetriever.getNextEventIdForOrganization(organizationId)(m2mContexts)
     _ = logger.debug("Next event id for organization {} -> {}", nextEvent.organizationId, nextEvent.eventId)
-    _ = dynamoService.put(organizationId, nextEvent.eventId, msg)
-  } yield ()
+    result <- dynamoService.put(organizationId, nextEvent.eventId, msg)
+    _ = logger.debug("Message {} was successfully written to dynamodb", msg.messageUUID.toString)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/notifier/service/impl/QueueHandler.scala]: I know this is to keep the same approach, but if possible let's avoid using this positional implementation 🙏

**Review by galales — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/notifier/service/impl/QueueHandler.scala [hunk 2]
```diff
@@ -44,11 +44,12 @@ class QueueHandler(
...
     m2mContexts = Seq(CORRELATION_ID_HEADER -> UUID.randomUUID().toString, BEARER -> m2mToken.serialized)
     organizationId <- getRecipientId(msg.payload)(m2mContexts)
-    _ = logger.debug("Organization id retrieved for message {} -> {}", msg.messageUUID, organizationId)
+    _ = logger.debug(s"Organization id retrieved for message  ${msg.messageUUID} -> $organizationId")
     nextEvent <- idRetriever.getNextEventIdForOrganization(organizationId)(m2mContexts)
-    _ = logger.debug("Next event id for organization {} -> {}", nextEvent.organizationId, nextEvent.eventId)
-    _ = dynamoService.put(organizationId, nextEvent.eventId, msg)
-  } yield ()
+    _ = logger.debug(s"Next event id for organization ${nextEvent.organizationId} -> ${nextEvent.eventId}")
+    result <- dynamoService.put(organizationId, nextEvent.eventId, msg)
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/notifier/service/impl/QueueHandler.scala:50]: Did we decide to make it blocking so?

Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/notifier/service/impl/QueueHandler.scala:50]: Yes, this is the job part that is not involved in the API flow.

Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/notifier/service/impl/QueueHandler.scala:50]: Add another argument in favor of this approach: a failure writing on dynamodb (what currently happened during my tests) is not intercepted, so we remove the message from the queue. :trollface:

**Review by TonioGela — APPROVED** [review]

**Pull request merged**